.github/workflows/rspec.yml

```diff
@@ -0,0 +1,32 @@
+name: RSpec
+on: push
+
+jobs:
+  rspec:
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby_version: ['2.7.6', '3.0.5', '3.1.4', '3.2.2']
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: wbari/start-mongoDB@v0.2
+        with:
+          mongoDBVersion: '4.2'
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby_version }}
+      - name: Cache Gems
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-aca_entities-gems-${{ hashFiles('**/Gemfile.lock') }}-${{ hashFiles('**/Gemfile' ) }}
+          restore-keys: |
+            ${{ runner.os }}-aca_entities-gems-${{ hashFiles('**/Gemfile.lock') }}-${{ hashFiles('**/Gemfile' ) }}
```

Comment:
You want to add the ruby version to the hash key for gems, it will make sure that gems aren't reused (and for the gems that have a C component, recompiled) for the different versions.

Check out https://github.com/ideacrew/event_source/commit/62b581802a6d191e7d96cba8c7b761a8d278e348#diff-1fadaba7942eb03a370984f465dc59789417e580083e4bcaa2d073959b1815be to see how.